c7n/resources/efs.py

```diff
@@ -227,3 +229,69 @@ def fetch_resources_lfc(self, resources):
             except client.exceptions.FileSystemNotFound:
                 continue
         return resources
+
+
+@ElasticFileSystem.filter_registry.register('check-secure-transport')
+class CheckSecureTransport(Filter):
+    """Find EFS that does not enforce secure transport
+
+    :Example:
+
+    .. code-block:: yaml
+
+     - name: efs-securetransport-check-policy
+       resource: efs
+       filters:
+         - check-secure-transport
+
+    To configure an EFS to enforce secure transport, set up the appropriate
+    Effect and Condition for its policy document. For example:
+
+    .. code-block:: json
+
+        {
+            "Sid": "efs-statement-b3f6b59b-d938-4001-9154-508f67707073",
+            "Effect": "Deny",
+            "Principal": { "AWS": "*" },
+            "Action": "*",
+            "Condition": {
+                "Bool": { "aws:SecureTransport": "false" }
+            }
+        }
+    """
+
+    schema = type_schema('check-secure-transport')
+    permissions = ('elasticfilesystem:DescribeFileSystemPolicy',)
+
+    def securetransport_check_policy(self, client, resource):
+        try:
+            resp = client.describe_file_system_policy(FileSystemId=resource['FileSystemId'])
+            if 'Policy' in resp:
+                policies = resp['Policy']
+        except Exception as e:
```

Comment:
nit: error handling can be simplified by catching on client.exceptions.PolicyNotFound

Reply:
@kapilt updated per feedback